zopkio/test_recipes.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2015 LinkedIn Corp.
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from datetime import time as dtime
+from dateutil import parser
+import os
+import threading
+import random
+import logging
+import time
+
+import zopkio.runtime as runtime
+
+def test_kill_recovery(kill_deployer, kill_deployer_process_id, restart_func_dict=None, verify_recovery=None, timeout=None):
+  """
+  A test recipe to kill a component in distributed system and verify system recovery once its brought back.
+  :param kill_deployer: The deployer one of whose process needs to be killed
```

Comment:
maybe consider passing in a dictionary or processe_ids to deployers to test multiple deployers at once